e2e/app/page/google-login.ts

```diff
@@ -118,32 +114,34 @@ export default class GoogleLoginPage extends BasePage {
         console.error('Google login button not found. ' + err);
         throw err;
       });
-      await this.clickAndWait(googleLoginButton);
+      await BaseElement.asBaseElement(this._page, googleLoginButton).clickAndWait();
 
       if (!user || user.trim().length === 0) {
         console.warn('Login user email: value is empty!!!')
       }
       await this.enterEmail(user);
+      await this._page.waitFor(1000); // reduce chances of getting Google login recaptcha
       await this.enterPassword(pwd);
       await this.submit();
     } catch (err) {
-      await takeScreenshot(this.page, 'FailedLoginPage');
-      await savePageToFile(this.page, 'FailedLoginPage');
-      throw err;
+      await takeScreenshot(this._page);
+      await savePageToFile(this._page);
+      throw new Error(err);
     }
 
-    try {
-      await this.waitUntilTitleMatch('Homepage');
-    } catch (e) {
-      // Handle "Enter Recovery Email" prompt if found exists
-      const recoverEmail = await this.page.$x('//input[@type="email" and @aria-label="Enter recovery email address"]');
-      if (recoverEmail.length > 0) {
-        await recoverEmail[0].type(config.contactEmail);
-        await Promise.all([
-          this.page.waitForNavigation(),
-          this.page.keyboard.press(String.fromCharCode(13)), // press Enter key
-        ]);
-      }
+    // Sometimes, user is prompted with "Enter Recovery Email" page. Handle the page if found.
+    await Promise.race([
+      waitUntilMatchPageTitle(this._page, 'Homepage', 30000),
+      this._page.waitForXPath('//input[@type="email" and @aria-label="Enter recovery email address"]', {visible: true, timeout: 30000}),
+    ]);
+    const elementHandles = await this._page.$x('//input[@type="email" and @aria-label="Enter recovery email address"]');
+    if (elementHandles.length > 0) {
+      await elementHandles[0].type(config.broadInstitutionEmail);
+      await Promise.all([
+        this._page.waitForNavigation(),
+        this._page.keyboard.press(String.fromCharCode(13)), // press Enter key
```

Comment:
Please define a constant for this and/or add a `hitEnter()` method to the base class.